src/Microsoft.SqlTools.ServiceLayer/Profiler/ProfilerService.cs

```diff
@@ -120,6 +121,39 @@ public void InitializeService(ServiceHost serviceHost)
             this.SessionMonitor.AddSessionListener(this);
         }
 
+        /// <summary>
+        /// Handle request to start a profiling session
+        /// </summary>
+        internal async Task HandleStartProfilingRequest(StartProfilingParams parameters, RequestContext<StartProfilingResult> requestContext)
+        {
+            await Task.Run(async () =>
```

Comment:
This was how the code was originally found in the profiler extension, I kept it as is to avoid disrupting functionality, let me know if this is needed or not. @alanrenmsft Do we still need to keep this async?

Reply:
this issue is not just with profiler handlers, it is a much wider problem that I am looking into. so it sounds good to me to keep things as is. I am planning to come up with something that will address all the issues at once.